BDArmory/Ammo/ModuleCASE.cs

```diff
@@ -451,23 +451,36 @@ void OnGUI()
                 }
             }
         }
-
+        float blastSim = 0f;
+        float simTimer = 0;
+        Color blastColor = Color.red;
         void DrawDetonationVisualization()
         {
             GetBlastRadius();
+            if (simTimer < 5)
```

Comment:
`OnGUI` gets called arbitrarily many times per frame (at least twice), so this timer isn't working as intended.
This patch should fix that and also automatically reset the animation whenever the `BD_AMMOBINS` option or the weapon alignment indicator gets toggled.
```diff
diff --git a/BDArmory/Ammo/ModuleCASE.cs b/BDArmory/Ammo/ModuleCASE.cs
index 2540b864c..90072592c 100644
--- a/BDArmory/Ammo/ModuleCASE.cs
+++ b/BDArmory/Ammo/ModuleCASE.cs
@@ -440,31 +440,28 @@ void OnDestroy()
 
         void OnGUI()
         {
-            if (HighLogic.LoadedSceneIsEditor && BDArmorySettings.BD_AMMOBINS)
+            if (HighLogic.LoadedSceneIsEditor)
             {
-                if (BDArmorySetup.showWeaponAlignment)
+                if (BDArmorySetup.showWeaponAlignment && BDArmorySettings.BD_AMMOBINS)
                     DrawDetonationVisualization();
-                else
+                else if (simStartTime > 0)
                 {
                     visSphere.SetActive(false);
                     visDome.SetActive(false);
+                    simStartTime = 0;
                 }
             }
         }
         float blastSim = 0f;
-        float simTimer = 0;
+        float simStartTime = 0;
+        float simTimer => Time.time - simStartTime;
         Color blastColor = Color.red;
         void DrawDetonationVisualization()
         {
             GetBlastRadius();
-            if (simTimer < 5)
-            {
-                simTimer += Time.fixedDeltaTime / 4;
-                blastSim = simTimer;
-                blastSim = Mathf.Clamp01(blastSim);
-                blastColor = Color.HSVToRGB((blastRadius * blastSim/ blastRadius) / 6, 1, 1);
-            }
-            else simTimer = 0;
+            if (simTimer > 5) simStartTime = Time.time;
+            blastSim = Mathf.Clamp01(simTimer);
+            blastColor = Color.HSVToRGB(blastRadius * blastSim / blastRadius / 6, 1, 1);
 
             switch (CASELevel)
             {
```